internal/ringbuffer.go

```diff
@@ -0,0 +1,119 @@
+package internal
+
+import (
+	"fmt"
+	"github.com/hazelcast/hazelcast-go-client/core"
+	"github.com/hazelcast/hazelcast-go-client/internal/common/collection"
+	. "github.com/hazelcast/hazelcast-go-client/internal/protocol"
+)
+
+type RingbufferProxy struct {
+	*partitionSpecificProxy
+}
+
+func newRingbufferProxy(client *HazelcastClient, serviceName *string, name *string) (*RingbufferProxy, error) {
+	parSpecProxy, err := newPartitionSpecificProxy(client, serviceName, name)
+	if err != nil {
+		return nil, err
+	}
+	return &RingbufferProxy{parSpecProxy}, nil
+}
+
+func (rp *RingbufferProxy) Capacity() (capacity int64, err error) {
+	request := RingbufferCapacityEncodeRequest(rp.name)
```

Comment:
In java it looks like there is a different implementation 
https://github.com/hazelcast/hazelcast/blob/master/hazelcast-client/src/main/java/com/hazelcast/client/proxy/ClientRingbufferProxy.java#L107

Reply:
It caches the capacity value so that client won't ask it server again to improve performance.